vault/resource_aws_secret_backend_role.go

```diff
@@ -97,8 +107,14 @@ func awsSecretBackendRoleWrite(d *schema.ResourceData, meta interface{}) error {
 		policy = d.Get("policy")
 	}
 
-	if policy == "" && len(policyARNs) == 0 {
-		return fmt.Errorf("either policy or policy_arn must be set.")
+	var roleARNs []string
+	roleARNsIfc := d.Get("role_arns")
+	for _, roleIfc := range roleARNsIfc.([]interface{}) {
+		roleARNs = append(roleARNs, roleIfc.(string))
+	}
+
+	if policy == "" && len(policyARNs) == 0 && len(roleARNs) == 0 {
```

Comment:
Not sure this check is needed.

Reply:
Are you talking about the whole `if { }` or just the `&& len(roleARNs) == 0` part?